router.go

```diff
@@ -192,3 +208,13 @@ func (m MapRouter) Choose(ctx context.Context, req *transport.Request) (transpor
 
 	return transport.HandlerSpec{}, yarpcerrors.Newf(yarpcerrors.CodeUnimplemented, "unrecognized procedure %q for service %q", req.Procedure, req.Service)
 }
+
+// Extract keys from service names map and return a formatted string
+func getAvailableServiceNames(svcMap map[string]struct{}, conjunction string) string {
+	serviceNames := reflect.ValueOf(svcMap).MapKeys()
```

Comment:
Let's always try to avoid reflection whenever possible. We don't need this. You can `range` over a map like a list to get the keys
```golang
for key,value := range serviceMap {
 ...
}
```
and since we don't care about the values (`struct{}`), we can just do
```golang
for key := range serviceMap {
 ...
}
```